### PR TITLE
Fix error message for staticmethod/classmethod order with validate_call

### PR DESCRIPTION
**Pull request by dmontagu**
closes https://github.com/pydantic/pydantic/issues/5329

Selected Reviewer: @samuelcolvin

### CODE REVIEW

File: pydantic/validate_call.py [hunk 1]
```diff
@@ -43,6 +43,14 @@ def validate_call(
...
 
     def validate(function: AnyCallableT) -> AnyCallableT:
+        if isinstance(function, classmethod):
+            raise TypeError(
+                'The `@classmethod` decorator should be applied after `@validate_call` (put `@classmethod` on top)'
+            )
+        elif isinstance(function, staticmethod):
+            raise TypeError(
+                'The `@staticmethod` decorator should be applied after `@validate_call` (put `@staticmethod` on top)'
+            )
```
Comment (dmontagu) [inline pydantic/validate_call.py:53]:
```suggestion
        if isinstance(function, (classmethod, staticmethod)):
            name = type(function).__name__
            raise TypeError(
                f'The `@classmethod` decorator should be applied after `@{name}` (put `@{name}` on top)'
            )
```
Could do this instead, not sure which is preferable.

**Comment by dmontagu** [comment]
please review

**Review by samuelcolvin — APPROVED** [review]
If that passes, LGTM.

**Review by samuelcolvin — CHANGES_REQUESTED** [review]
tests failing.

**Comment by dmontagu** [comment]
please review

**Comment by samuelcolvin** [comment]
TYVM

**Pull request merged**